### PR TITLE
MINOR: [R][DOCS] Add sections about `Scalar$create()`

### PR DESCRIPTION
**Pull request by eitsupi**
Copied almost the same content as `Array$create()`.
However, only the behavior when a data.frame is passed is special, so an example has been added.

### CODE REVIEW

**Review by thisisnic — COMMENTED** [review]
Thanks for helping expand the docs!  One small comment, and perhaps a follow-up one to come about the explanation of passing in a data.frame to Scalar$create, but let me check something and get back to you on that!

File: r/R/scalar.R [hunk 1]
```diff
@@ -44,6 +58,8 @@
 #' Scalar$create(404)
 #' # If you pass a vector into Scalar$create, you get a list containing your items
 #' Scalar$create(c(1, 2, 3))
+#' # If you pass a data.frame with multiple columns into Scalar$create, you get the first row of the data.frame
+#' Scalar$create(mtcars)
```
Comment (thisisnic) [inline r/R/scalar.R:62]:
```suggestion
```
Let's leave this out; it's very atypical usage, and not something we'd expect users to need to do.

Comment (eitsupi) [reply r/R/scalar.R]: I have no strong reason to insist that this be done, so I will remove it.

Comment (thisisnic) [reply r/R/scalar.R]: Cheers!  And thanks in general for looking into these docs issues - these kinds of things make for a much better experience for our users.

**Review by thisisnic — APPROVED** [review]
Wonderful, thanks!

**Pull request closed without merging**